apps/ledger-live-desktop/src/renderer/screens/nft/Collections/Collections.tsx

```diff
@@ -71,11 +72,14 @@ const Collections = ({ account }: Props) => {
     [account.id, history],
   );
 
-  const { nfts, fetchNextPage, hasNextPage } = useNftGalleryFilter({
+  const { enabled, hideSpamCollection } = useHideSpamCollection();
+
+  const { nfts, fetchNextPage, hasNextPage } = useCheckNftAccount({
     nftsOwned: account.nfts || [],
     addresses: account.freshAddress,
     chains: [account.currency.id],
     threshold: isThresholdValid(thresold) ? Number(thresold) : 75,
+    action: enabled ? hideSpamCollection : undefined,
```

Comment:
you can do it dynamically in one shot `...(enabled && {action: hideSpamCollection})`